graph/src/data/graphql/ext.rs

```diff
@@ -266,10 +273,78 @@ impl DirectiveFinder for Field {
             .iter()
             .find(|directive| directive.name.eq(name))
     }
+
+    fn is_derived(&self) -> bool {
+        let is_derived = |directive: &Directive| directive.name.eq("derivedFrom");
+
+        self.directives.iter().any(is_derived)
+    }
 }
 
 impl DirectiveFinder for Vec<Directive> {
     fn find_directive(&self, name: &str) -> Option<&Directive> {
         self.iter().find(|directive| directive.name.eq(&name))
     }
+
+    fn is_derived(&self) -> bool {
+        let is_derived = |directive: &Directive| directive.name.eq("derivedFrom");
+
+        self.iter().any(is_derived)
+    }
+}
+
+#[cfg(test)]
+mod directive_finder_tests {
+    use graphql_parser::parse_schema;
+
+    use super::*;
+
+    const SCHEMA: &str = "
+    type BuyEvent implements Event @derivedFrom(field: \"buyEvent\") {
+        id: ID!,
+        transaction: Transaction! @derivedFrom(field: \"buyEvent\")
+    }";
```

Comment:
Just minor note:
As far as I can tell, the `derivedFrom` directive is applied to fields, not to types. 
Thus, I'd suggest removing the directive from the type and focusing the test only on fields marked by it.